public/js/index.js

```diff
@@ -5,13 +5,14 @@ var defaultExtraKeys = {
       cm.setOption("fullScreen", !cm.getOption("fullScreen"));
     },
     "Esc": function(cm) {
-      if (cm.getOption("fullScreen")) cm.setOption("fullScreen", false);
+       if (cm.getOption('keyMap').substr(0, 3) === 'vim') return CodeMirror.Pass;
```

Comment:
`cm.getOption('keyMap')` will return `vim-insert` In the insert mode;

![image](https://cloud.githubusercontent.com/assets/3684597/14763540/f250f34e-09ca-11e6-8163-e4a4b71bfebb.png)